Kernel/FileSystem/MountFile.cpp

```diff
@@ -28,83 +29,99 @@ ErrorOr<NonnullLockRefPtr<MountFile>> MountFile::create(FileSystemInitializer co
     // to ensure this never happens.
     VERIFY(!(flags & MS_BIND));
     VERIFY(!(flags & MS_REMOUNT));
-    auto mount_specific_data_buffer = TRY(KBuffer::try_create_with_size("Mount Specific Data"sv, PAGE_SIZE, Memory::Region::Access::ReadWrite, AllocationStrategy::AllocateNow));
-    return TRY(adopt_nonnull_lock_ref_or_enomem(new (nothrow) MountFile(file_system_initializer, flags, move(mount_specific_data_buffer))));
+    return TRY(adopt_nonnull_lock_ref_or_enomem(new (nothrow) MountFile(file_system_initializer, flags)));
 }
 
-MountFile::MountFile(FileSystemInitializer const& file_system_initializer, int flags, NonnullOwnPtr<KBuffer> mount_specific_data)
+MountFile::MountFile(FileSystemInitializer const& file_system_initializer, int flags)
     : m_flags(flags)
     , m_file_system_initializer(file_system_initializer)
 {
-    m_mount_specific_data.with_exclusive([&](auto& our_mount_specific_data) {
-        our_mount_specific_data = move(mount_specific_data);
-        memset(our_mount_specific_data->data(), 0, our_mount_specific_data->size());
-    });
 }
 
 MountFile::~MountFile() = default;
 
 ErrorOr<void> MountFile::ioctl(OpenFileDescription&, unsigned request, Userspace<void*> arg)
 {
-    return m_mount_specific_data.with_exclusive([&](auto& our_mount_specific_data) -> ErrorOr<void> {
+    return m_filesystem_specific_options.with_exclusive([&](auto& filesystem_specific_options) -> ErrorOr<void> {
+        auto user_mount_specific_data = static_ptr_cast<MountSpecificFlag const*>(arg);
+        auto mount_specific_data = TRY(copy_typed_from_user(user_mount_specific_data));
+        if ((mount_specific_data.value_type == MountSpecificFlag::ValueType::SignedInteger || mount_specific_data.value_type == MountSpecificFlag::ValueType::UnsignedInteger) && mount_specific_data.value_length != 8)
+            return EDOM;
+
+        Syscall::StringArgument user_key_string { reinterpret_cast<char const*>(mount_specific_data.key_string_addr), static_cast<size_t>(mount_specific_data.key_string_length) };
+        auto key_string = TRY(Process::get_syscall_name_string_fixed_buffer<MOUNT_SPECIFIC_FLAG_KEY_STRING_MAX_LENGTH>(user_key_string));
+
         switch (request) {
+        case MOUNT_IOCTL_DELETE_MOUNT_SPECIFIC_FLAG: {
+            if (!filesystem_specific_options.remove(key_string.representable_view()))
+                return Error::from_errno(ENXIO);
```

Comment:
I believe `NXIO` is primarily for network operations. Not sure if `ENOENT` or `ESRCH` is more appropriate (but given that we use `EEXIST` in the `SET` case, the antithesis to that would be `ENOENT`). Could maybe just be a silent fail?